### PR TITLE
express-delete

### PR DESCRIPTION
**Pull request by dannyandreev**
image

### CODE REVIEW

File: express-delete/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+};
+
+const app = express();
+
+// console.log(grades)
+
+app.delete('/api/grades/:id', (req, res) => {
+
+  delete grades['12'];
```
Comment (scbowler) [inline express-delete/index.js:30]: You don't want to hard code the `id` here. You should be using `req.params.id` to get the `id` to delete

File: express-delete/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  // console.log(grades)
+  const gradeArray = [];
+  for (const id in grades) {
+    gradeArray.push(grades[id]);
+  }
+
```
Comment (scbowler) [inline express-delete/index.js:37]: There is no need for a loop here. There should be a separate `GET` endpoint that returns the grades in an array.

File: express-delete/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  // res.sendStatus(204)
+  res.json(gradeArray);
+  res.send(res.params);
```
Comment (scbowler) [inline express-delete/index.js:40]: You can't send multiple things back from an endpoint, you can only send one thing back. And for delete you should just be sending back the status code `204`. To send back just a status code use `res.sendStatus(204)`

File: express-delete/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+const app = express();
+
+// console.log(grades)
+
```
Comment (scbowler) [inline express-delete/index.js:27]: You should have a get endpoint here that returns an array of grades. Just like the one from the previous exercise.

**Review by scbowler — CHANGES_REQUESTED** [review]
Some updates needed, see below comments for details. Let me know if you have any questions.

**Comment by dannyandreev** [comment]
image

File: express-delete/index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  // console.log('delete')
+  const idToDelete = req.params.id;
+  // console.log(idToDelete)
+
+  delete grades[idToDelete];
+  // console.log(grades)
+  res.status(204).send();
```
Comment (scbowler) [inline express-delete/index.js:45]: You could also use `res.sendStatus(204)` here if you are just sending a status.

**Review by scbowler — APPROVED** [review]
Great job implementing all the feedback. Looks good! 👍 

1 FYI comment, no changes required

**Pull request merged**